scripts/dts/edtlib.py

```diff
@@ -954,6 +954,9 @@ def _prop_val(self, name, prop_type, required, default):
         # prop_type:
         #   Property type from binding (a string like "int")
         #
+        # deprecated:
```

Comment:
Curious: is there going to be any situation where a property is `deprecated` and also `required`?

Reply:
> Curious: is there going to be any situation where a property is `deprecated` and also `required`?

There had better not

Reply:
Added a check to error if both `deprecated` and `required` are True